### PR TITLE
Fix: Display avatar after fetching user data

### PR DESCRIPTION
**Pull request by jackyshows**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [x] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?
(https://github.com/goetzrobin/spartan/issues/338)

## What is the new behavior?
The avatar component no longer has an error state; it now only handles a loaded state. It correctly updates the user avatar when the src attribute changes, without always showing the fallback avatar. Inside the `@HostListener('error')` method, it sets the loaded state to false, and the canShow flag only checks if the avatar is loaded or not.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/ui/tooltip/brain/src/lib/brn-tooltip-content.component.ts [hunk 1]
```diff
@@ -219,6 +219,11 @@ export class BrnTooltipContentComponent implements OnDestroy {
 		const tooltip = this._tooltip?.nativeElement;
 		if (!tooltip || !this._isBrowser) return;
 		this._renderer2.setStyle(tooltip, 'visibility', isVisible ? 'visible' : 'hidden');
+		if (isVisible){
+			this._renderer2.removeStyle(tooltip, 'display');
+		}else{
+			this._renderer2.setStyle(tooltip, 'display', 'none');
+		}
```
Comment (ajitzero) [inline libs/ui/tooltip/brain/src/lib/brn-tooltip-content.component.ts:226]: This seems unrelated, please rebase from the `main` branch!

**Review by ajitzero — CHANGES_REQUESTED** [review]
LGTM. Thanks for the PR!

File: libs/ui/tooltip/brain/src/lib/brn-tooltip-content.component.ts (hunk 1 above)
Comment (jackyshows) [reply libs/ui/tooltip/brain/src/lib/brn-tooltip-content.component.ts:226]: @ajitzero this is my first time working on an open source project. You're right, I'm not sure why this commit is here, and I don't know how to fix it. I tried to rebase from the main branch, but it doesn't seem to work; I'm surely doing something wrong. Feel free to delete this PR, and I'll delete my forked repo and create a new, clean PR. Alternatively, if you prefer, you can create a new one yourself.

Comment (ajitzero) [reply libs/ui/tooltip/brain/src/lib/brn-tooltip-content.component.ts:226]: No worries! git can get confusing with forks and whatnot. I see you have 3 extra commits in this PR https://github.com/goetzrobin/spartan/pull/342/commits - I'm not sure what could have caused this either. Could you please try removing these last three commits?

Try these CLI commands:
```shell
# go to your repo/fork's branch with the changes
git checkout main

# remove the last N (here, 3) commits: git reset --hard HEAD~N
git reset --hard HEAD~3

# Update your fork on GitHub forcibly and overwrite it.
# We don't do this for actual projects, but is common for cleaning up PRs.
git push --force
```
To be cautious though, I would suggest running the reset command without the `--hard` flag. This will only "undo" the changes and keep them available locally. You can then decide what to skip manually.
